src/components/Gif/Gif.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {MasonryItem, Overlay, ImageWrapper, LinkStyled, GifImg, GifTitle} from "./Gif.styled";
+
+const getPreviewGifUrl = (gif) => {
+  if (gif.images && gif.images.original) {
+    return gif.images.original.url;
+  } else {
+    return '';
+  }
+};
+
+const handleFavButton = (gif, addGif, removeGif, isFavorited) => {
+  if (!isFavorited) {
+    addGif(gif);
+  } else {
+    removeGif(gif.id);
+  }
+};
+
+export const Gif = ({ gif, fullWidth, addGif, removeGif, isFavorited, isInDetail }) => (
```

Comment:
when having a lot of props "destructed", is better to destruct them inside the function to increase readability